File: media10progressiva.py
s = 0

for i in range(10):
    if i < 1:
        n1 = int(input("Digite um número: "))
        s += n1
    else:
        n1 = int(input("Digite outro número: "))
        s += n1
        s /= 2
        print(f'A média de agora é {s}')

print(f"A média é {s}")

File: media10progressiva.c
#include <stdio.h>

int main() {
    float sm = 0;
    for (int i = 0; i < 10; i++)
    {
        float n;
        if (i < 1)
        {
            printf("Digite um número:\n");
            scanf("%f", &n);
            sm += n;
        }
        else
        {
            printf("Digite outro número: \n");
            scanf("%f", &n);
            sm = (sm + n) / 2;
        }
        printf("A média é: %.1f\n", sm);
    
    }
return 0;

    
}